### ISSUE TITLE
SP_DIR is wrong in test phase in noarch: python build

### ISSUE THREAD

**Issue by isuruf** [conda/conda-build#2977]
### Actual Behavior
```
SP_DIR = /home/conda/feedstock_root/build_artifacts/sagenb_1529477105378/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python2.7/site-packages/
```

but test env is as follows,
```
## Package Plan ##

  environment location: /home/conda/feedstock_root/build_artifacts/sagenb_1529477105378/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p


The following NEW packages will be INSTALLED:

    attrs:             18.1.0-py_1          conda-forge
    automat:           0.7.0-py35_0         conda-forge
    babel:             2.6.0-py35_0         conda-forge
    ca-certificates:   2018.4.16-0          conda-forge
    certifi:           2018.4.16-py35_0     conda-forge
    click:             6.7-py_1             conda-forge
    constantly:        15.1.0-py_0          conda-forge
    defusedxml:        0.5.0-py35_0         conda-forge
    flask:             0.11.1-py_1          conda-forge
    flask-autoindex:   0.6-py_0             local      
    flask-babel:       0.11.2-py35_0        conda-forge
    flask-oldsessions: 0.10-py_0            local      
    flask-openid:      1.2.5-py35_0         conda-forge
    flask-silk:        0.2-py_0             local      
    future:            0.16.0-py35_0        conda-forge
    hyperlink:         17.3.1-py_0          conda-forge
    incremental:       17.5.0-py_0          conda-forge
    itsdangerous:      0.24-py_2            conda-forge
    jinja2:            2.10-py35_0          conda-forge
    libgcc-ng:         7.2.0-hdf63c60_3     defaults   
    markupsafe:        1.0-py35_0           conda-forge
    ncurses:           5.9-10               conda-forge
    openssl:           1.0.2o-0             conda-forge
    python:            3.5.5-1              conda-forge
    python3-openid:    3.1.0-py35_0         conda-forge
    pytz:              2018.4-py_0          conda-forge
    readline:          7.0-0                conda-forge
    sagenb:            1.0.1-py_0           local      
    setuptools:        39.2.0-py35_0        conda-forge
    six:               1.11.0-py35_1        conda-forge
    sqlite:            3.20.1-2             conda-forge
    tk:                8.6.7-0              conda-forge
    twisted:           17.5.0-py35_0        defaults   
    werkzeug:          0.14.1-py_0          conda-forge
    xz:                5.2.3-0              conda-forge
    zlib:              1.2.11-h470a237_3    conda-forge
    zope.interface:    4.5.0-py35h470a237_0 conda-forge
```
### Expected Behavior
```
SP_DIR=/home/conda/feedstock_root/build_artifacts/sagenb_1529477105378/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python2.7/site-packages
```
### Steps to Reproduce

See https://circleci.com/gh/isuruf/staged-recipes/617

##### `Output of conda info`

```
     active environment : base
    active env location : /opt/conda
            shell level : 1
       user config file : /home/conda/.condarc
 populated config files : /home/conda/.condarc
          conda version : 4.5.4
    conda-build version : 3.10.5
         python version : 3.6.5.final.0
       base environment : /opt/conda  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/linux-64
                          https://repo.anaconda.com/pkgs/pro/noarch
          package cache : /opt/conda/pkgs
                          /home/conda/.conda/pkgs
       envs directories : /opt/conda/envs
                          /home/conda/.conda/envs
               platform : linux-64
             user-agent : conda/4.5.4 requests/2.18.4 CPython/3.6.5 Linux/4.4.0-96-generic centos/6.9 glibc/2.12
                UID:GID : 1001:1001
             netrc file : None
           offline mode : False
```

**Comment by msarahan** [comment]
please update conda-build.  A possible fix was done in 3.10.6: https://github.com/conda/conda-build/releases/tag/3.10.6

**Comment by isuruf** [comment]
Thanks. I didn't realize staged-recipes was not updating conda-build. Fixed that
